modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -156,19 +93,31 @@ def inform(self, new_data: list[tuple[int, int, int]]) -> Generator[int, None, N
             if detection_idx_end <= self.leftover_data_points:
                 continue
 
-            # trigger_id doesn't always start from 0
             if self.previous_trigger_id is None:
-                # if no previous trigger exists, always trigger retraining
-                triggered = True
+                triggered = True  # if no previous trigger exists, always trigger retraining
+                drift_results: dict[str, MetricResult] = {}
```

Comment:
This is the logic at the beginning, right? Should we have something like having the first trigger time or amount based? Right now not sure when exactly the first trigger would fire, maybe it would make sense to synchronize the first trigger to then have a reference point for drift?

Reply:
Good point, I just took the current solution so far, but that makes sense.

Reply:
Currently, it just takes the first batch acc. to `detection_interval_data_points` and train on it. That's fair for now, maybe we want sth. timestamp based in the future.